### PR TITLE
Crash Web Browser

### PR DESCRIPTION
**Pull request by theck13**
### Fix
Add detecting if the device has a browser app installed and enabled before launching an activity with a browser intent.  If the device does not have a browser or it is disabled, the app will crash as described in https://github.com/Automattic/simplenote-android/issues/1043.  When no browser is detected, a dialog is shown with instructions as well as a button to copy the URL.  A message is shown when copying the URL succeeds or fails.  See the screenshots below for illustration.

crash_web_browser

### Test
These changes are best tested on Simplenote by pointing your local Simplenote repository to your local Simperium repository.  Contact me for details.  In order to get the ***No Browser Detected*** dialog to be shown, all browser apps must be uninstalled or disabled on the device before performing the following steps.  In order to test the ***Reset*** button in the ***Error*** dialog, change your Simplenote password to more than four and less than eight characters.

0. Clear app data.
1. Tap ***Sign Up*** button.
2. Tap ***Terms and Conditions*** link.
3. Notice ***No Browser Detected*** dialog is shown.
4. Tap ***OK*** button in dialog.
5. Tap back arrow in top app bar.
6. Tap ***Log In*** button.
7. Tap ***Log In with Email*** button.
8. Tap ***Forgot password?*** link.
9. Notice ***No Browser Detected*** dialog is shown.
10. Tap ***OK*** button in dialog.
11. Enter email address in ***Email*** field.
12. Notice ***Log In*** button is disabled.
13. Enter password of more than four and less than eight characters in ***Password*** field.
14. Notice ***Log In*** button is enabled after four characters are input.
15. Tap ***Log In*** button.
16. Notice ***Error*** dialog with password requirements is shown.
17. Tap ***Reset*** button in dialog.
18. Notice ***No Browser Detected*** dialog is shown.
19. Tap ***Copy URL*** button in dialog.
20. Notice ***URL copied to clipboard*** message is shown.

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

### CODE REVIEW

File: Simperium/src/main/java/com/simperium/android/CredentialsActivity.java [hunk 1]
```diff
@@ -324,6 +330,23 @@ private void clearPassword() {
...
+    private void copyToClipboard(String url) {
+        Context context = new ContextThemeWrapper(CredentialsActivity.this, getTheme());
+
+        try {
+            ClipboardManager clipboard = (ClipboardManager) getSystemService(Context.CLIPBOARD_SERVICE);
+            ClipData clip = ClipData.newPlainText(getString(R.string.app_name), url);
+
+            if (clipboard != null) {
+                clipboard.setPrimaryClip(clip);
+            }
```
Comment (AmandaRiu) [inline Simperium/src/main/java/com/simperium/android/CredentialsActivity.java]: If the clipboard is `null` then we don't actually copy the data to the clipboard, but we still show a snack message stating it was successful. Is that intentional?

**Review by AmandaRiu — COMMENTED** [review]
@theck13 Ran through the tests and all good 👍 Just have one code comment for your review.

File: Simperium/src/main/java/com/simperium/android/CredentialsActivity.java (hunk 1 above)
Comment (theck13) [reply Simperium/src/main/java/com/simperium/android/CredentialsActivity.java]: Good point!  That is not intentional.  I updated the logic to show the failure message when `clipboard` is `null` in a0e5c53.

**Review by AmandaRiu — APPROVED** [review]
LGTM! 🚢

**Pull request merged**